app/src/main/java/com/example/henzoshimada/feeltrip/MainScreen.java

```diff
@@ -82,32 +85,42 @@ public boolean onNavigationItemSelected(@NonNull MenuItem item) {
             switch (item.getItemId()) {
                 case R.id.navigation_home:
                     Log.d("Mytag","Tapped on home");
+                    FeelTripApplication.setFrag("main");
                     fragment = new homeFragmment();
                     ft.replace(R.id.fragent_frame,fragment);
                     //ft.addToBackStack(null);
                     ft.commit();
-                    return true;
+                    break;
                 case R.id.navigation_profile:
                     Log.d("Mytag","Tapped on profile");
+                    FeelTripApplication.setFrag("profile");
                     fragment = new profileFragment();
                     ft.replace(R.id.fragent_frame,fragment);
                     //ft.addToBackStack(null);
                     ft.commit();
-                    return true;
+                    break;
                 case R.id.navigation_map:
                     Log.d("Mytag","Tapped on map");
+                    FeelTripApplication.setFrag("map");
                     fragment = new mapFragment();
                     ft.replace(R.id.fragent_frame,fragment);
                     //ft.addToBackStack(null);
                     ft.commit();
-                    return true;
+                    break;
                 default:
                     fragment = new homeFragmment();
                     ft.replace(R.id.fragent_frame,fragment);
                     ft.addToBackStack(null);
                     ft.commit();
-                    return true;
+                    break;
             }
+//            Intent i = getIntent();
+//            i.addFlags(Intent.FLAG_ACTIVITY_NO_HISTORY);
+//            finish();
+//            startActivity(i);
+            //onPause(); // Refreshes the current activity without calling onCreate
+            onResume();
+            return true;
```

Comment:
This optimizes the number of times we query from Elasticsearch, thereby reducing our load times by a whole extra query lookup. That's a lot.